File: javaapi/src/main/java/org/team2363/helixtrajectory/TrajectorySample.java
package org.team2363.helixtrajectory;

public class TrajectorySample {
    public final double ts, x, y, heading, vx, vy, omega;

    public TrajectorySample(double ts, double x, double y,
            double heading, double vx, double vy, double omega) {
        this.ts = ts;
        this.x = x;
        this.y = y;
        this.heading = heading;
        this.vx = vx;
        this.vy = vy;
        this.omega = omega;
    }
}

File: helixtrajectorycpp/src/cpp/trajectory/TrajectorySample.cpp
#include "trajectory/TrajectorySample.h"

namespace helixtrajectory {

    TrajectorySample::TrajectorySample(double intervalDuration, double x, double y, double heading)
            : intervalDuration(intervalDuration), x(x), y(y), heading(heading) {
    }

    std::ostream& operator<<(std::ostream& stream, const TrajectorySample& sample) {
        return stream << "{\"x\": " << sample.x
                << ", \"y\": " << sample.y
                << ", \"heading\": " << sample.heading
                << "}";
    }
}